### ISSUE TITLE
Unclosed Logback Threads

### ISSUE THREAD

**Issue by PvanHengel** [splunk/splunk-library-javalogging#66]
I am trying to use the splunk-library-javalogging library in a command line running spring boot application, without the library the app gracefully shuts down, however after adding the splunk logger the app never terminates, as far as I can tell this is due to hanging threads.  

Also note we are using the HttpEventCollectorLogbackAppender

The test is quite simple to replicate this, just run the below springboot app.

```
@SpringBootApplication
public class SampleApp implements CommandLineRunner {

	private static final Logger LOGGER = LoggerFactory.getLogger(ArchiveController.class);
	
	public static void main(String[] args) {
		SpringApplication.run(SampleApp.class, args);
	}

	@Override
	public void run(String... args) throws Exception {

		LOGGER.info("Starting App");
		// Do Some Work here

		LOGGER.info("Ending App");

System.out.println( Thread.activeCount() ); // Returns 2 for app without splunk, returns 11 for apps with..

		// the app should close without a call to system.exit(0)

	}

}
```

important parts of pom:

```
<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>1.5.8.RELEASE</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>

<dependency>
			<groupId>com.splunk.logging</groupId>
			<artifactId>splunk-library-javalogging</artifactId>
			<version>1.5.2</version>
		</dependency>

```

**Comment by tdhellmann** [comment]
Thanks for including the test case! We'll take a look.

**Comment by izrobin** [comment]
This is also a problem for me. I'm adding the logback appender in my Dropwizard logging configuration, and the thread is blocking when JVM is trying to shut down.

**Comment by shakeelmohamed** [comment]
Can you see if this is fixed in the latest v1.6.1?

**Comment by mattiaslevin** [comment]
I get the same problem in 1.7.1

**Comment by Delai21** [comment]
Same problem here in 1.7.3.
We're currently using the following workaround:

```java
@SpringBootApplication
public class SomeTask {

  public static void main(String[] args) {
    System.exit(SpringApplication.exit(SpringApplication.run(SomeTask.class, args)));
  }
}
```

**Comment by zenmoto** [comment]
This should be resolved by release 1.8.0.
